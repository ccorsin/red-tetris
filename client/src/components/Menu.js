import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { regex } from "../utils/regex";

const Menu = () => {
  const [isValid, setIsValid] = useState(false);
  const [player, setPlayer] = useState("PLAYER");
  const [roomNb, setRoomNb] = useState("");
  let history = useHistory();

  const checkRoom = value => {
      if (regex.room.exec(value)) {
          setIsValid(true);
          setRoomNb(value);
      } else {
          setIsValid(false);
          setRoomNb("");
      }
  };

  const checkPlayer = value => {
    if (regex.username.exec(value) && regex.room.exec(value)) {
      setIsValid(true);
      setPlayer(value);
    } else {
      setIsValid(false);
      setPlayer("PLAYER");
    }
  };

  const goToRoom = () => {
    if (isValid) {
      history.push("/room/#" + roomNb + "[" + player + "]");
      setPlayer("PLAYER");
      setRoomNb("");
      setIsValid(false);
    }
  }

  return (
    <div>
      <div>
        <input
          type="text"
          placeholder="PLAYER"
          onChange={e => checkPlayer(e.target.value)}
        />
      </div>
      <div>
        <input
          type="text"
          placeholder="ROOM"
          onChange={e => checkRoom(e.target.value)}
        />
        <span role="alert" id="error-name-required">
          Only numbers.
        </span>
      </div>
      <div>
        <button onClick={e => goToRoom(true)} disabled={!isValid}>
          PLAY
        </button>
      </div>
    </div>
  );
};

export default Menu;
