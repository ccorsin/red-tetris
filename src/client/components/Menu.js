import React, { useState } from "react";
import { useHistory } from "react-router-dom";


const Menu = () => {
  let history = useHistory();
  const reg = /^([\d]+)$/;
  const [isValid, setisValid] = useState(false);
  const [player, setPlayer] = useState("PLAYER");
  const [roomNb, setRoomNb] = useState("");

  const checkRoom = value => {
      if (reg.exec(value)) {
          setisValid(true);
          setRoomNb(value);
      } else {
          setisValid(false);
          setRoomNb("");
      }
  };

  const goToRoom = () => {
    if (isValid) {
      history.push("/playground/#" + roomNb + "[" + player + "]");
      socket.emit("room", roomNb, player);
      // setPath("/playground/#" + roomNb + "[" + player + "]");
      setPlayer("PLAYER");
      setRoomNb("");
      setisValid(false);
    }
  }
    
  return (
    <div>
      <div>
        <input
          type="text"
          placeholder="PLAYER"
          onChange={e => playerName(e.target.value)}
        />
      </div>
      <div>
        <input
          type="text"
          placeholder="ROOM"
          onChange={e => checkRoom(e.target.value)}
        />
        <span role="alert" id="error-name-required">
          Only numbers.
        </span>
      </div>
      <div>
        <button onClick={e => goToRoom(true)} disabled={!isValid}>
          PLAY
        </button>
      </div>
    </div>
  );
};

export default Menu;
