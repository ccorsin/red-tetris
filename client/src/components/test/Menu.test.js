import React from 'react';
import { Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import { useHistory } from "react-router-dom";
import Menu from '../Menu';
import toJson from "enzyme-to-json"

describe('<Menu/> Component', () => {
  it('rendering correctly Menu', () => {
    const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
    let wrapper = shallow(
      <Router history={historyMock}>
        <Menu/>
      </Router>
    )
    expect(wrapper.html()).toMatchSnapshot()
  })
})