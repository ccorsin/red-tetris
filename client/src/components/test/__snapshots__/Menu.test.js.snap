// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> Component rendering correctly Menu 1`] = `"<div class=\\"sc-AxjAm dQSdpZ\\"><div class=\\"sc-AxirZ cfHwMP\\"><span class=\\"menu_label\\">TAKE A ROOM NUMBER</span><div class=\\"menu_input\\"><input type=\\"text\\" placeholder=\\"PLAYER\\"/><span class=\\"error-number-required\\">ONLY LETTERS OR NUMBERS</span></div><div class=\\"menu_input\\"><input type=\\"text\\" placeholder=\\"ROOM\\"/><span class=\\"error-number-required\\">ONLY NUMBERS</span></div><div class=\\"menu_input\\"><button class=\\"button-to-game\\" disabled=\\"\\">PLAY</button></div></div></div>"`;
